runtime/src/epoch_stakes.rs

```diff
@@ -196,7 +245,7 @@ pub(crate) fn split_epoch_stakes(
                 versioned_epoch_stakes.insert(
                     epoch,
                     VersionedEpochStakes::Current {
-                        stakes: Stakes::<Stake>::from(stakes.clone()),
```

Comment:
This is where the expensive clone and stakes conversion was happening. This PR delays the conversion until serialization and even goes one step further by borrowing the stake state instead of copying it to reduce memory usage during said serialization